src/Simplify.cpp

```diff
@@ -346,6 +346,16 @@ class Simplify : public IRMutator {
             // outer cast is narrower, the inner cast can be
             // eliminated.
             expr = mutate(Cast::make(op->type, cast->value));
+        } else if (cast &&
+                   (op->type.code() == Type::Int || op->type.code() == Type::UInt) &&
+                   (cast->type.code() == Type::Int || cast->type.code() == Type::UInt) &&
+                   op->type.bits() < cast->type.bits() &&
+                   op->type.bits() < op->value.type().bits()) {
```

Comment:
I was thinking that the case where these are equal is covered by other rules... but I'll change it anyways.